admin/cluster_setup.py

```diff
@@ -193,14 +217,25 @@ def main(reactor, args, base_path, top_level):
                 print("Didn't finish creating the cluster.")
                 runner.stop_cluster(reactor)
             else:
-                print("The following variables describe the cluster:")
                 environment_variables = get_trial_environment(cluster)
+                environment_strings = list()
                 for environment_variable in environment_variables:
-                    print("export {name}={value};".format(
-                        name=environment_variable,
-                        value=shell_quote(
-                            environment_variables[environment_variable]),
-                    ))
+                    environment_strings.append(
+                        "export {name}={value};\n".format(
+                            name=environment_variable,
+                            value=shell_quote(
+                                environment_variables[environment_variable]
+                            ),
+                        )
+                    )
+                environment = ''.join(environment_strings)
+                print("The following variables describe the cluster:")
+                print(environment)
```

Comment:
Since we're already mixing IO with computation, there's no reason to build `environment_strings`, join it, then print it. Instead, we might as well print the "export ..." string as we figure it out.